addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoUtil.java

```diff
@@ -0,0 +1,290 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.io.neeo.NeeoConstants;
+import org.openhab.io.neeo.internal.models.NeeoDeviceType;
+import org.openhab.io.neeo.internal.serialization.AtomicStringTypeAdapter;
+import org.openhab.io.neeo.internal.serialization.ChannelUIDSerializer;
+import org.openhab.io.neeo.internal.serialization.ThingUIDSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * Various utility functions used by the NEEO Connector
+ *
+ * @author Tim Roberts - initial contribution
+ */
+public class NeeoUtil {
+
+    /**
+     * Builds and returns a {@link GsonBuilder}. The GsonBuilder has adapters registered for {@link ThingUID},
+     * {@link ChannelUID}
+     * and for atomic string references
+     *
+     * @return a non-null {@link GsonBuilder} to use
+     */
+    public static GsonBuilder createGsonBuilder() {
+        final GsonBuilder gsonBuilder = new GsonBuilder();
+
+        gsonBuilder.registerTypeAdapter(ThingUID.class, new ThingUIDSerializer());
+        gsonBuilder.registerTypeAdapter(ChannelUID.class, new ChannelUIDSerializer());
+        gsonBuilder.registerTypeAdapter(new TypeToken<AtomicReference<String>>() {
+        }.getType(), new AtomicStringTypeAdapter());
+        return gsonBuilder;
+    }
+
+    /**
+     * Builds and returns a {@link Gson} from {@link #createGsonBuilder()}
+     *
+     * @return the non-null {@link Gson} to use
+     */
+    public static Gson createGson() {
+        return createGsonBuilder().create();
+    }
+
+    /**
+     * Gets the servlet url for the given brain id.
+     *
+     * @param brainId the non-empty brain id
+     * @return the servlet url
+     */
+    public static String getServletUrl(String brainId) {
+        requireNotEmpty(brainId, "brainId cannot be empty");
+        return NeeoConstants.WEBAPP_PREFIX + "/" + brainId;
+    }
+
+    /**
+     * Gets the unique identifier of this openHAB instance
+     *
+     * @return the unique id
+     */
+    public static String getUniqueId() {
+        return InstanceUUID.get();
+    }
+
+    /**
+     * Decodes the passed UTF-8 String using an algorithm that's compatible with
+     * JavaScript's <code>decodeURIComponent</code> function. Returns
+     * <code>null</code> if the String is <code>null</code>.
+     *
+     * @param s The UTF-8 encoded String to be decoded
+     * @return the decoded String
+     */
+    public static String decodeURIComponent(String s) {
+        if (s == null) {
+            return null;
+        }
+
+        String result = null;
+
+        try {
+            result = URLDecoder.decode(s, StandardCharsets.UTF_8.name());
+        }
+
+        // This exception should never occur.
+        catch (UnsupportedEncodingException e) {
```

Comment:
Please align / join with the try and if you need the comment maybe add it inside the catch